### PR TITLE
Fix linter errors

### PR DESCRIPTION
**Pull request by iaguis**
https://github.com/koalaman/shellcheck/wiki/SC2048

We get a https://github.com/koalaman/shellcheck/wiki/SC2016
but we do mean literal backquotes.

### CODE REVIEW

**Comment by 2opremio** [comment]
Please don't modify tools directory in-place (it's a git tree).

Also, I believe that `generate_latest_map` if anything, should live on `/extras` (we may not really need it though).

Would you mind creating a PR to remove it in https://github.com/weaveworks/build-tools and another one to bump the git-tree (and optional add the script to extras) in the repository?

Thanks

**Comment by 2opremio** [comment]
@iaguis ping

**Comment by iaguis** [comment]
Sorry for letting this slip through the cracks.

It seems `generate_latest_map` is already removed from build-tools. I updated this PR with a build-tools bump and re-added the script to `extras/`.

**Comment by iaguis** [comment]
Hm, and now I get tons of shellcheck errors

**Comment by iaguis** [comment]
Not sure what the problem is now. The error is not very useful:

```
$ cd $SRCDIR; make RM= lint
scope: run shfmt -i 4 -w scope

Makefile:105: recipe for target 'lint' failed

make: *** [lint] Error 1

make: Leaving directory '/go/src/github.com/weaveworks/scope'

make: *** [lint] Error 2

cd $SRCDIR; make RM= lint returned exit code 2
```

**Comment by iaguis** [comment]
The problem were shfmt issues. PTAL.

File: bin/release [hunk 1]
```diff
@@ -26,10 +23,10 @@ infer_release_type() {
 setup() {
     # Ensure we have exactly one annotated tag pointing at HEAD
     HEAD_TAGS=$(git tag --points-at HEAD)
-	TAG_COUNT=$(echo $(echo $HEAD_TAGS | wc -w)) # mac hack
+    TAG_COUNT=$(echo "$HEAD_TAGS" | wc -w) # mac hack
```
Comment (alban) [inline bin/release]: what is this "mac hack"? Was the double "echo" done for this mac hack?

Comment (iaguis) [inline bin/release]: cc @tomwilkie

Comment (alban) [inline bin/release]: https://github.com/weaveworks/scope/commit/58a1528ba6f22ae9a7e84ee1eb22287e987734dd#commitcomment-20505530

We probably need to add a comment to tell shellcheck not to remove the double echo then...

Comment (2opremio) [inline bin/release]: Yep, seems like it.

Comment (iaguis) [inline bin/release]: Re-added the double echo

**Comment by 2opremio** [comment]
Thanks a lot for this @iaguis , it must have been a pain ...

**Comment by 2opremio** [comment]
Crap, I missed there are linter fixes in `tools/lint`, which should had been done in build-tools

**Comment by iaguis** [comment]
Oh noes, I missed that too 😞

**Comment by 2opremio** [comment]
Again, please remember that tools is a subtree

**Comment by 2opremio** [comment]
I will use the file version from build-tools, if the linter keeps bitching we can fix it there.

File: extras/build_on_circle.sh [hunk 2]
```diff
@@ -2,4 +2,4 @@
 
 set -e
 
-./in_parallel.sh "make RM=" $(find . -maxdepth 2 -name *.go -printf "%h\n" | sort -u | sed -n 's/\.\/\(.*\)/\1\/\1/p')
+./in_parallel.sh "make RM=" "$(find . -maxdepth 2 -name "./*.go" -printf "%h\n" | sort -u | sed -n 's/\.\/\(.*\)/\1\/\1/p')"
```
Comment (alban) [inline extras/build_on_circle.sh]: It seems like it is the wrong fix ^
Shouldn't we add an "ignore" for the linter check? 
@iaguis @alepuccetti @schu

**Pull request merged**